### PR TITLE
Magento 2.3.4 could not get category image name - 27168

### PR DESCRIPTION
**Pull request by RHapani**
### Description (*)
Fixed Category Image Url Bug.

### Steps to reproduce (*)

1. Create a Category and add an image to that category.
2. I need to display the category image of few categories based on id in homepage. So i have try to 
use the code in frontend as below,

`$category->getImageUrl()`
`$category->getImage()`

But it returns the tmp path image of that category.

I have even tried using the above codes for getting category image name in 
**vendor/magento/module-catalog/view/frontend/templates/category/image.phtml** but end up with getting wrong result.

### Expected result (*)
1.$category->getImageUrl() - It should return full path of category image({baseUrl}/pub/media/catalog/category/M2CAD.png)

2.$category->getImage()- It should return only the name of category image(M2CAD.png)

### Actual result (*)
1.Both the code returns temporary(tmp) path image of the category ({baseUrl}/pub/media/catalog/tmp/category/M2CAD.png)
2.Couldn't get the filename of the category image separately.

### Fixed Issues (if relevant)

1. magento/magento2[#27168](https://github.com/magento/magento2/issues/27168): Magento 2.3.4 could not get category image name

### Manual testing scenarios (*)

1. Get specific category object
2. print $categoryObj->getImageUrl();

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by miteshdhaduk** [comment]
@lbajsarowicz Why merging is blocked? There is no issue with code customization.

**Comment by drpayyne** [comment]
Irrespective of the failing tests, isn’t `$usingPathRelativeToBase` redundant here? Wouldn’t directly returning the condition check work?

File: app/code/Magento/Catalog/Model/Category/Attribute/Backend/Image.php [hunk 1]
```diff
@@ -196,8 +196,9 @@ private function fileResidesOutsideCategoryDir($value)
         if (!$baseMediaDir) {
             return false;
         }
+        $usingPathRelativeToBase = strpos($fileUrl, $baseMediaDir) === 0;
```
Comment (diazwatson) [inline app/code/Magento/Catalog/Model/Category/Attribute/Backend/Image.php]:
```suggestion
        
```

File: app/code/Magento/Catalog/Model/Category/Attribute/Backend/Image.php [hunk 2]
```diff
@@ -196,8 +196,9 @@ private function fileResidesOutsideCategoryDir($value)
...
 
-        return strpos($fileUrl, $baseMediaDir) !== false;
+        return $usingPathRelativeToBase;
```
Comment (diazwatson) [inline app/code/Magento/Catalog/Model/Category/Attribute/Backend/Image.php]:
```suggestion
        return strpos($fileUrl, $baseMediaDir) === 0;
```

**Review by diazwatson — CHANGES_REQUESTED** [review]
Hi @RHapani thanks for your contribution!

This is basically the change @drpayyne is suggesting. Please have a look.

BTW: I have heard this is your first contribution so welcome aboard 🎉!!!

File: app/code/Magento/Catalog/Model/Category/Attribute/Backend/Image.php [hunk 3]
```diff
@@ -196,9 +196,9 @@ private function fileResidesOutsideCategoryDir($value)
         if (!$baseMediaDir) {
             return false;
         }
-        $usingPathRelativeToBase = strpos($fileUrl, $baseMediaDir) === 0;
+
```
Comment (diazwatson) [inline app/code/Magento/Catalog/Model/Category/Attribute/Backend/Image.php]:
```suggestion
```

**Review by diazwatson — CHANGES_REQUESTED** [review]

**Review by diazwatson — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @diazwatson, thank you for the review.
[ENGCOM-7089](https://jira.corp.magento.com/browse/ENGCOM-7089) has been created to process this Pull Request
:eight_spoked_asterisk: @diazwatson, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by magento-engcom-team** [comment]
@RHapani thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Review by engcom-Alfa — CHANGES_REQUESTED** [review]
Hi @RHapani .
     Due to [Magento Definition of Done](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing_dod.html) all code must be covered by tests. Please cover your fix by automate test.
Thanks!

**Comment by VladimirZaets** [comment]
@RHapani , I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for collaboration

**Pull request closed without merging**